apiary.apib

```diff
@@ -1545,6 +1545,106 @@ Lists tables in a given bucket.
                 }
             ]
 
+## Create table definition [/v2/storage/buckets/{bucket_id}/tables-definition]
+
+### Create new table definition [POST]
+
+**EXPERIMENTAL**
+
+Creates a table with typed columns. *Synapse only.*
++ Parameters
+    + bucket_id (required) - The bucket Id
+
++ Attributes
+    + name (required) - New table name
+    + primaryKeysNames[] (string) - has to be subset of column names
+    + columns[] (required)
+       + name
+       + definition (object)
+            + type (required) - datatype of this column, choose one from https://bit.ly/3uZjN60
+            + length (string) - optional, default: depends on type
+            + nullable (boolean) - optional, default: true
+            + default (string) - optional, default value for column
+    + distribution
+        + type (required, enum[string])
+            + Members
+                + HASH - requires just on column in `distributionColumnsNames`
+                + REPLICATE
+                + ROUND_ROBIN
+        + distributionColumnsNames[] (required) - has to be subset of column names
+
++ Request (application/json)
+    + Headers
+
+            X-StorageApi-Token: your_token
+
+    + Body
+
+            {
+                "name": "table_name",
+                "primaryKeysNames": [],
+                "columns": [
+                    {
+                        "name": "id",
+                        "definition": {
+                            "type": "INT",
+                            "length": "10",
```

Comment:
```suggestion
                            "length": null,
```
pre int to nieje validna hodnota
<img width="665" alt="Screenshot 2021-05-17 at 16 16 27" src="https://user-images.githubusercontent.com/6448364/118503787-3c6a2c80-b72b-11eb-9a55-a81e1d3a1c24.png">